### PR TITLE
Upgrade to react 18 from react 17

### PR DESCRIPTION
**Pull request by jinmayamashita**
@mvn-binhnguyen-hn @mvn-luanvu-hn
I upgraded from React 18.1.0 and react-dom 18.1.0 and types.

Also, I changed to update our rendering root to the new client so it will support the new features.

As an aside:
React 18 related libraries may be good to manage automatically with renovate etc.

### CODE REVIEW

File: src/main.tsx [hunk 1]
```diff
@@ -15,10 +15,14 @@ const prepare = async (): Promise<void> => {
 
 prepare().then(() => {
   useI18next();
-  render(
+
+  const container = document.getElementById("root");
+  if (!container) throw new Error("container element is missing.");
+
+  const root = createRoot(container);
```
Comment (jinmayamashita) [inline src/main.tsx:22]: I Changed to use `createRoot` according to [documentation](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis).

**Comment by jinmayamashita** [comment]
@n-igarashi

I would like to install a [renovate](https://www.whitesourcesoftware.com/free-developer-tools/renovate/) available on the Github App to handle automatic updates of other libraries.
Is it possible to Install the Github App in the repository? 😄

File: src/main.tsx (hunk 1 above)
Comment (mvn-binhnguyen-hn) [reply src/main.tsx:22]: I think we can use `createRoot(container!)` so we don't need to throw Error when `container` is not existed
image

Comment (jinmayamashita) [reply src/main.tsx]: @mvn-binhnguyen-hn We have accepted your suggestion! [dd956a3](https://github.com/monstar-lab-oss/reactjs-boilerplate/pull/28/commits/dd956a31d2be2d04a4e658bf86edbc6cd04dd78f)

**Review by mvn-binhnguyen-hn — APPROVED** [review]

**Pull request merged**